### DISCUSSION TITLE
tutorial add supported wallet

### DISCUSSION THREAD

**Discussion by qlixes** [wevm/wagmi#179] [Q&A] [answered]
Dear @tmm and team, I had newbie question. 
Please inform, how to add supported wallet on wagmi.js ?
what i must todo and confirm about supported wallet ?
example adding trust wallet, tokenpocket, binance chain etc.
Thank you a lot

**Comment by tmm** [comment, marked as answer]
Hi! Wallets are set up with [connectors](https://wagmi-xyz.vercel.app/guides/connectors). wagmi has built-in connectors for MetaMask, WalletConnect, and WalletLink, but you can also create your own if you need support for specific wallets. (Note: WalletConnect supports many [different wallets](https://walletconnect.com/registry/wallets), including Trust Wallet.)

**Reply by qlixes** [reply]
I'm sorry for late reply. if do you mind, can you inform me how to use WalletConnectConnector for each supported wallet ? on my mind, from WalletConnectConnector, I could create button for each supported by WalletConnectConnector.
Thank you a lot

**Reply by qlixes** [reply]
Please let me know how to get wallet's information, because I'm still confuse about each wallet. because each of wallet had etherum network or binance network.
Thank you a lot

**Comment by qlixes** [comment]
Dear @tmm and team, I have some questions :
1. on Math Wallet, it's using InjectedConnector, but with window.etherum.isMathWallet was true, what i must todo for adding Math Wallet support in InjectedConnector ?
2. on Binance Wallet, it's using InjectedConnector, but with window.BinanceChain was true, what i must todo for adding Binance Chain Wallet support in InjectedConnector ? 
3. it's compatible using wagmi feature while add Connector from @binance-chain/bsc-connector package ?
4. As mention, WalletConnect support multiple wallet, How to get instance connect foreach supported wallet from WalletConnectConnector ?
Thank you a lot
